avm/res/compute/virtual-machine/main.bicep

```diff
@@ -0,0 +1,767 @@
+  'Desktop Virtualization Power On Contributor': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '489581de-a3bd-480d-9518-53dea7416b33')
+  'Desktop Virtualization Power On Off Contributor': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '40c5ff49-9181-41f8-ae61-143b0e78555e')
+  'Desktop Virtualization Virtual Machine Contributor': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'a959dbd1-f747-45e3-8ba6-dd80f235f97c')
+  'DevTest Labs User': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '76283e04-6283-4c54-8f91-bcf1374a3c64')
+  'Disk Backup Reader': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '3e5e47e6-65f7-47ef-90b5-e5dd4d455f24')
+  'Disk Pool Operator': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '60fc6e62-5479-42d4-8bf4-67625fcc2840')
+  'Disk Restore Operator': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'b50d9833-a0cb-478e-945f-707fcc997c13')
+  'Disk Snapshot Contributor': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '7efff54f-a5b4-42b5-a1c5-5411624893ce')
+  Owner: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '8e3af657-a8ff-443c-a75c-2fe8c4bcb635')
+  Reader: subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'acdd72a7-3385-48ef-bd42-f606fba81ae7')
+  'Role Based Access Control Administrator (Preview)': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'f58310d9-a9f6-439a-9e8d-f62e7b41a168')
+  'User Access Administrator': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '18d7d88d-d35e-4fb5-a5c3-7773c20a72d9')
+  'Virtual Machine Administrator Login': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '1c0163c0-47e6-4577-8991-ea5c82e286e4')
+  'Virtual Machine Contributor': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', '9980e02c-c2be-4d73-94e8-173b1dc7cf3c')
+  'Virtual Machine User Login': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'fb879df8-f326-4884-b1cf-06f3ad86be52')
+  'VM Scanner Operator': subscriptionResourceId('Microsoft.Authorization/roleDefinitions', 'd24ecba3-c1f4-40fa-a7bb-4588a071e8fd')
+}
+
+resource avmTelemetry 'Microsoft.Resources/deployments@2023-07-01' = if (enableTelemetry) {
+  name: '46d3xbcp.res.compute-virtualmachine.${replace('-..--..-', '.', '-')}.${substring(uniqueString(deployment().name, location), 0, 4)}'
+  properties: {
+    mode: 'Incremental'
+    template: {
+      '$schema': 'https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#'
+      contentVersion: '1.0.0.0'
+      resources: []
+      outputs: {
+        telemetry: {
+          type: 'String'
+          value: 'For more information, see https://aka.ms/avm/TelemetryInfo'
+        }
+      }
+    }
+  }
+}
+
+module vm_nic 'br/public:avm/res/network/network-interface:0.2.2' = [for (nicConfiguration, index) in nicConfigurations: {
```

Comment:
Note: This module is missing the NIC+PIP deployment module from CARML: https://github.com/Azure/ResourceModules/blob/3e858dc2f82af7f5fe1864d803d127a077328d76/modules/compute/virtual-machine/main.bicep#L340

Reply:
@AlexanderSehr what should I add?

Reply:
My ask would be to recover the [logic](https://github.com/Azure/ResourceModules/blob/main/modules/compute/virtual-machine/modules/nested_networkInterface.bicep) that was provided by the original CARML module which was capable of not just deploying n NICs but also n Public IPs for each NIC configuration. 
Was this removed on purpose? I know I pointed this change in design (i.e., removal) out before, but maybe there was a reason afterall.

Reply:
Will look into it. I guess I forgot. Sorry. Will keep you posted.

Reply:
@AlexanderSehr maybe I am not understanding, but the NIC is now a separate module that I created prior to the VM and it is used by the VM. What is missing?

Reply:
Hey @rahalan that I'm not questioning and is exactly what you should do. But the original VM module never referenced the NIC module directly from the `main.bicep`, but instead an intermediate `nested_networkInterface.bicep` template that did a couple other things.
So my ask is to please recover that template and in it reference the external modules - just as we did in CARML (with the one difference that those were local file references).

Reply:
Hi @AlexanderSehr . After reviewing the code, I think you are asking for a change in the NIC module, not the VM one. Plus the NIC module already has the capability to deploy PIPs. Still not sure what needs to be changed.

Reply:
@AlexanderSehr . Done. [![avm.res.compute.virtual-machine](https://github.com/rahalan/bicep-registry-modules/actions/workflows/avm.res.compute.virtual-machine.yml/badge.svg?branch=users%2Frahalan%2FAlignVmToAVM2)](https://github.com/rahalan/bicep-registry-modules/actions/workflows/avm.res.compute.virtual-machine.yml)